### PR TITLE
Add docs sections

### LINKED ISSUE #23: Docs sections
It would be nice if we could break the docs into one page per component.

Resources:

* <https://github.com/styleguidist/react-styleguidist/blob/master/examples/sections/styleguide.config.js>

### PR DESCRIPTION
**Pull request by lukeschunk**
# Is there a better way?

## Change Type

* [ ] Feature
* [x] Chore
* [ ] Bug Fix

## Change Level

* [ ] major
* [ ] minor
* [x] patch

## Further Information (screenshots, links, etc)

Resolves #23

### CODE REVIEW

File: .styleguide/sectionHeading.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  const sectionNameClasses = cx(classes.sectionName, {
+    [classes.isDeprecated]: deprecated,
+  })
+
+  return (
+    <div
+      style={{
+        backgroundColor: headingLevel === 1 ? colors.action : null,
+        padding: headingLevel === 1 ? 30 : null,
+        color: headingLevel === 1 ? colors.white : null,
```
Comment (lukeschunk) [inline .styleguide/sectionHeading.js]: basically i just copied their component https://github.com/styleguidist/react-styleguidist/blob/master/src/rsg-components/SectionHeading/SectionHeadingRenderer.js and added some inline styles... 🤦‍♂️

File: .styleguide/sectionHeading.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+import Heading from 'rsg-components/Heading'
+import Styled from 'rsg-components/Styled'
+import { colors } from '../src/theme.js'
+
+function SectionHeadingRenderer({
```
Comment (zacanger) [inline .styleguide/sectionHeading.js]: Use a function expression.

File: .styleguide/sectionHeading.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  classes,
+  children,
+  toolbar,
+  id,
+  href,
+  depth,
+  deprecated,
+}) {
+  const headingLevel = Math.min(6, depth)
+  const sectionNameClasses = cx(classes.sectionName, {
```
Comment (zacanger) [inline .styleguide/sectionHeading.js]: This can go, unless you want to start bringing in CSS.

File: styleguide.config.js [hunk 4]
```diff
@@ -62,7 +73,7 @@ module.exports = {
     '**/theme.js',
   ],
   webpackConfig,
-  theme,
+  theme: {},
```
Comment (zacanger) [inline styleguide.config.js]: Revert this line

File: .styleguide/sectionHeading.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
+import React from 'react'
```
Comment (zacanger) [inline .styleguide/sectionHeading.js]: Use kebab-case file names.

File: .styleguide/sectionHeading.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        <a href={href} className={sectionNameClasses}>
+          {children}
+        </a>
+      </Heading>
+      <div className={classes.toolbar}>{toolbar}</div>
+    </div>
+  )
+}
+
+const styles = ({ color, space }) => ({
```
Comment (zacanger) [inline .styleguide/sectionHeading.js]: You could use styled-components here.

File: .styleguide/sectionHeading.js [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+  },
+  isDeprecated: {
+    color: color.light,
+    '&, &:hover': {
+      textDecoration: 'line-through',
+    },
+  },
+})
+
+SectionHeadingRenderer.propTypes = {
```
Comment (zacanger) [inline .styleguide/sectionHeading.js]: Use Flow.

File: package.json [hunk 8]
```diff
@@ -57,6 +57,7 @@
     "babel-preset-env": "1.7.0",
     "babel-preset-react": "6.24.1",
     "babel-preset-stage-2": "6.24.1",
+    "classnames": "2.2.6",
```
Comment (zacanger) [inline package.json]: No need for this lib if we use css-in-js.

File: .styleguide/sectionHeading.js (hunk 5 above)
Comment (TomPridham) [reply .styleguide/sectionHeading.js]: also, `import * as React from 'react'`

**Pull request merged**